colossus/src/main/scala/colossus/core/Connection.scala

```diff
@@ -194,7 +195,14 @@ private[core] abstract class Connection(val id: Long, val key: SelectionKey, _ch
   def handleRead(data: DataBuffer)(implicit time: Long) = {
     _lastTimeDataReceived = time
     myBytesReceived += data.size
-    handler.receivedData(data)
+    try {
+      handler.receivedData(data)
+    }
+    catch {
+      case parsingException:ParseException => {
+        throw new ParseException(s"error in request $remoteAddress", parsingException)
```

Comment:
I'm not really a fan of this.  It seems too application-specific for the core layer, also kind of weird that you're rethrowing the same exception but just adding the remote address.  

I think it would be better to somehow add support for `ParseException` handling (and other "pre-request-handling" errors) into the service layer.  That way we could have a default behavior but allow the service to override handling to do things like adding more info to the message.  This would be different, though, from the existing `onError`, since that is for handling uncaught exception during request processing, while this happens before request processing even starts.  Perhaps it can be somehow unified with connection closed errors, since that happens outside of request processing as well.  I will create an issue for this.